File: backend/Models/Enumerations/TicketStatus.cs
// TicketStatus.cs
namespace backend.Models.Enumerations;

using System.ComponentModel;

public enum TicketStatus
{
    [Description("New")]
    New = 0,

    [Description("On Review")]
    OnReview = 1,

    [Description("Support Will Contact You")]
    SupportWillContactYou = 2,

    [Description("In Progress")]
    InProgress = 3,

    [Description("Closed")]
    Closed = 4,

    [Description("Cancelled")]
    Cancelled = 5
}

File: frontend/src/utils/Tickets/ticketStatus.js
export const getStatusStyle = (status) => {
  switch (status) {
    case 0:
      return "bg-blue-100 text-blue-700";

    case 1:
      return "bg-yellow-100 text-yellow-700";

    case 2:
      return "bg-purple-100 text-purple-700";

    case 3:
      return "bg-orange-100 text-orange-700";

    case 4:
      return "bg-green-100 text-green-700";

    case 5:
      return "bg-red-100 text-red-700";

    default:
      return "bg-gray-100 text-gray-700";
  }
};

export const getStatusLabel = (status) => {
  switch (status) {
    case 0:
      return "New";

    case 1:
      return "On Review";

    case 2:
      return "Support Will Contact You";

    case 3:
      return "In Progress";

    case 4:
      return "Closed";

    case 5:
      return "Cancelled";

    default:
      return "Unknown";
  }
};
